### ISSUE TITLE
max backward can underflow

### ISSUE THREAD

**Issue by chenyuxyz** [tinygrad/tinygrad#12296]
```
from tinygrad import Tensor
t = Tensor.ones(70000, dtype="half", requires_grad=True).contiguous()
t.max().backward()
print(t.grad.tolist()[:10])
```
gives `[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]`

https://github.com/tinygrad/tinygrad/blob/476a2a0a962fa8a145c612f63b72eeb4dc3a84f3/tinygrad/gradient.py#L10 uses the gradient dtype for counting

**Comment by wp4032** [comment]
Hi @chenyuxyz, can you please check PR https://github.com/tinygrad/tinygrad/pull/12321 for the fix above?

**Comment by vznh** [comment]
is this still relevant? looks like 3 attempts & happy to take
